### PR TITLE
Update node version and use github-action for codecov

### PR DESCRIPTION
**Pull request by mmorel-35**
Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

## Which problem is this PR solving?
- Fix updates failures because of outdated node 10 version, for example in here : https://github.com/jaegertracing/jaeger-ui/actions/runs/3186395354/jobs/5196939882 

## Short description of the changes
- Update node version and use github-action for codecov

### CODE REVIEW

**Comment by yurishkuro** [comment]
> Fix upates failures because of outdated node 10 version

could you please elaborate which failures you're referring to?

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -15,17 +15,11 @@ jobs:
...
+          node-version: '16'
       - run: yarn install --frozen-lockfile
       - run: yarn lint
       - run: yarn build
       - run: yarn coverage
       - name: Upload coverage to codecov.io
-        run: |
-          yarn global add codecov
-          codecov
+        uses: codecov/codecov-action@v3
```
Comment (yurishkuro) [inline .github/workflows/unit-tests.yml:25]:
```suggestion
        uses: codecov/codecov-action@v3
        fail_ci_if_error: true
        verbose: true
```

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -15,17 +15,11 @@ jobs:
...
       - uses: actions/setup-node@v3
         with:
-          node-version: "10"
-      - name: install yarn
-        run: |
-          curl -o- -L https://yarnpkg.com/install.sh | bash
-          echo $HOME/.yarn/bin >>"$GITHUB_PATH"
-          echo $HOME/.config/yarn/global/node_modules/.bin >>"$GITHUB_PATH"
+          cache: yarn
+          node-version: '16'
```
Comment (yurishkuro) [inline .github/workflows/unit-tests.yml:19]: need to add it to `.nvmrc`

Comment (yurishkuro) [reply .github/workflows/unit-tests.yml:19]: Also, need to change it in the main `jaeger` repo:
```
# ~/dev/jaegertracing/jaeger
$ grep -rn node .github
.github/workflows/ci-release.yml:25:    - uses: actions/setup-node@v3
.github/workflows/ci-release.yml:27:        node-version: '10'
.github/workflows/ci-docker-build.yml:27:    - uses: actions/setup-node@v3
.github/workflows/ci-docker-build.yml:29:        node-version: '10'
.github/workflows/ci-all-in-one-build.yml:26:    - uses: actions/setup-node@v3
.github/workflows/ci-all-in-one-build.yml:28:        node-version: '10'
```

**Comment by mmorel-35** [comment]
> could you please elaborate which failures you're referring to?
I found this one, to start with : https://github.com/jaegertracing/jaeger-ui/actions/runs/3186395354/jobs/5196939882

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (mmorel-35) [reply .github/workflows/unit-tests.yml:19]: I created this this PR : https://github.com/jaegertracing/jaeger/pull/3957

**Review by yurishkuro — DISMISSED** [review]

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**